tensorflow_datasets/image/cartoonset.py

```diff
@@ -0,0 +1,150 @@
+  def __init__(self, data, **kwargs):
+    """Constructs a CartoonsetConfig.
+    Args:
+      data: `str`, one of `_DATA_OPTIONS`.
+      **kwargs: keyword arguments forwarded to super.
+    """
+    if data not in _DATA_OPTIONS:
+      raise ValueError("data must be one of %s" % _DATA_OPTIONS)
+
+    super(CartoonsetConfig, self).__init__(**kwargs)
+    self.data = data
+
+
+class Cartoonset(tfds.core.GeneratorBasedBuilder):
+  """CartoonSet is a collection of random, 2D cartoon avatar images."""
+
+  VERSION = tfds.core.Version('0.1.0')
+
+  BUILDER_CONFIGS = [
+      CartoonsetConfig(
+          name='cartoonset10k',
+          description="A collection of random, 10000 2D cartoon avatar images",
+          version="0.1.0",
+          data='cartoonset10k',
+      ),
+      CartoonsetConfig(
+          name='cartoonset100k',
+          description="A collection of random, 100000 2D cartoon avatar images",
+          version="0.1.0",
+          data='cartoonset100k',
+      )
+  ]
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict(
+            {"image": tfds.features.Image(shape=_CARTOON_IMAGE_SHAPE),
+             "eye_angle": tfds.features.ClassLabel(num_classes=3),
+             "eye_lashes": tfds.features.ClassLabel(num_classes=2),
+             "eye_lid": tfds.features.ClassLabel(num_classes=2),
+             "chin_length": tfds.features.ClassLabel(num_classes=3),
+             "eyebrow_weight": tfds.features.ClassLabel(num_classes=2),
+             "eyebrow_shape": tfds.features.ClassLabel(num_classes=14),
+             "eyebrow_thickness": tfds.features.ClassLabel(num_classes=4),
+             "face_shape": tfds.features.ClassLabel(num_classes=7),
+             "facial_hair": tfds.features.ClassLabel(num_classes=15),
+             "hair": tfds.features.ClassLabel(num_classes=111),
+             "eye_color": tfds.features.ClassLabel(num_classes=5),
+             "face_color": tfds.features.ClassLabel(num_classes=11),
+             "hair_color": tfds.features.ClassLabel(num_classes=10),
+             "glasses": tfds.features.ClassLabel(num_classes=12),
+             "glasses_color": tfds.features.ClassLabel(num_classes=7),
+             "eye_slant": tfds.features.ClassLabel(num_classes=3),
+             "eyebrow_width": tfds.features.ClassLabel(num_classes=3),
+             "eye_eyebrow_distance": tfds.features.ClassLabel(num_classes=3)}),
+        supervised_keys=None,
+        # Homepage of the dataset for documentation
+        urls=["https://google.github.io/cartoonset/"],
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    # There is no predefined train/val/test split for this dataset.
+    path = dl_manager.manual_dir
+    if not tf.io.gfile.exists(path):
+      msg = 'You must download the dataset files manually and place them in: '
```

Comment:
You must also set MANUAL_DOWNLOAD_INSTRUCTIONS field (see other datasets like c4)